zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/CamundaExporterIT.java

```diff
@@ -489,4 +492,126 @@ private ExporterResourceProvider mockResourceProvider(
+      final var decisionEntity =
+          new ImportPositionEntity()
+              .setPartitionId(partitionId)
+              .setAliasName(ImportValueType.DECISION.getAliasTemplate())
+              .setCompleted(false);
+
+      clientAdapter.index(
+          decisionEntity.getId(), importPositionIndex.getFullQualifiedName(), decisionEntity);
+
+      clientAdapter.refresh();
+
+      controller.runScheduledTasks(Duration.ofMinutes(1));
+
+      // then
+      final var record =
+          factory.generateRecord(
+              ValueType.AUTHORIZATION,
+              r -> r.withBrokerVersion("8.7.0").withTimestamp(System.currentTimeMillis()));
+
+      camundaExporter.export(record);
+
+      // then
+      assertThat(controller.getPosition()).isEqualTo(-1);
+      verify(controller, never()).updateLastExportedRecordPosition(eq(record.getPosition()), any());
+
+      final var authHandler =
+          getHandlers(config).stream()
+              .filter(handler -> handler.getHandledValueType().equals(record.getValueType()))
+              .filter(handler -> handler.handlesRecord(record))
+              .findFirst()
+              .orElseThrow();
+      final var recordId = authHandler.generateIds(record).getFirst();
+
+      assertThat(
+              clientAdapter.get(recordId, authHandler.getIndexName(), authHandler.getEntityType()))
+          .isNull();
+    }
+
+    @TestTemplate
+    void shouldFlushIfImportersAreCompleted(
+        final ExporterConfiguration config, final SearchClientAdapter clientAdapter)
+        throws IOException {
+      // given
+      config.getIndex().setShouldWaitForImporters(true);
+      final var context = spy(getContextFromConfig(config));
+      doReturn(partitionId).when(context).getPartitionId();
+      camundaExporter.configure(context);
+      camundaExporter.open(controller);
+
+      final var importPositionIndex = importPositionIndex(config);
+
+      // mark all import position documents as completed which signals all record readers as
+      // completed
+      for (final var type : ImportValueType.values()) {
+        final var entity =
+            new ImportPositionEntity()
+                .setPartitionId(partitionId)
+                .setAliasName(type.getAliasTemplate())
+                .setCompleted(true);
+        clientAdapter.index(entity.getId(), importPositionIndex.getFullQualifiedName(), entity);
+      }
+
+      controller.runScheduledTasks(Duration.ofMinutes(1));
+
+      // when
+      final var record =
+          factory.generateRecord(
+              ValueType.AUTHORIZATION,
+              r -> r.withBrokerVersion("8.7.0").withTimestamp(System.currentTimeMillis()));
+
+      camundaExporter.export(record);
+
+      // then
+      assertThat(controller.getPosition()).isEqualTo(record.getPosition());
+      verify(controller, times(1))
+          .updateLastExportedRecordPosition(eq(record.getPosition()), any());
```

Comment:
🔧 we might be able to directly assert the position

Reply:
Not sure what you mean here I already directly assert the position, do you mean remove the verify call?

Reply:
Lol nevermind. I thought we can do `eq(position` but we already do 🤦🏼 Sorry.